core/source_map.rs

```diff
@@ -92,7 +94,9 @@ impl<G: SourceMapGetter> SourceMapper<G> {
     let getter = self.getter.as_ref();
     let maybe_source_map =
       self.maps.entry(file_name.to_owned()).or_insert_with(|| {
-        let raw_source_map = getter.and_then(|it| it.get_source_map(file_name));
+        let raw_source_map = None
+          .or_else(|| self.ext_source_maps.get(file_name).cloned())
```

Comment:
I worry this clone here could be expensive. Can we get the SourceMap from the slice here?

Reply:
I don't think that is necessary for a few reasons:
1. `SourceMapGetter::get_source_map` already returns a `Vec<v8>` object, which most likely already performs a clone internally
2. The resulting `SourceMap` gets cached, therefore this code runs only once per source map.
3. The code is (as far as I know) already on a cold path that is only reached while displaying an error.

Reply:
Regardless, I'm pretty sure we're doing an extra clone that doesn't need to happen.

Let's see if we can clean this logic up a bit -- can you try to use `if let Some(...)` instead of `or_else` chaining?

I'll take a look but I think we may be able to change the `SourceMapGetter` to avoid the current required cloning, that we can land in a future PR.